ui-src/src/cells/holo-vault/components/profile/fieldMapper.tsx

```diff
@@ -53,24 +145,71 @@ class FieldMapper extends React.Component<Props, State> {
   constructor (props: Props) {
     super(props)
     this.state = {
+      suggestions: [],
+      personaAutovalue: props.selectedPersona.name,
+      selectedPersona: props.selectedPersona,
+      fieldAutovalue: '',
       expansionPanelOpen: false,
       mappedPersona: props.selectedPersona,
       mappedField: props.field.name
     }
   }
 
+  static getDerivedStateFromProps (nextProps: Props, prevState: State) {
```

Comment:
I'm still a little bit concerned about our heavy use of getDerivedStateFromProps, only because the official react docs state that there are only a few rare cases where it should be used.

That said it is easy to understand and seems to work pretty well so we'll keep it for now.